barretenberg/cpp/src/barretenberg/vm2/constraining/relations/tx.test.cpp

```diff
@@ -0,0 +1,489 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+#include "barretenberg/vm2/common/avm_inputs.hpp"
+#include "barretenberg/vm2/common/aztec_constants.hpp"
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/constraining/flavor_settings.hpp"
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/columns.hpp"
+#include "barretenberg/vm2/generated/relations/execution.hpp"
+#include "barretenberg/vm2/testing/fixtures.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/testing/public_inputs_builder.hpp"
+#include "barretenberg/vm2/tracegen/lib/lookup_builder.hpp"
+#include "barretenberg/vm2/tracegen/precomputed_trace.hpp"
+#include "barretenberg/vm2/tracegen/public_inputs_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using tx = bb::avm2::tx<FF>;
+using lookup_read_phase_table_relation = bb::avm2::lookup_tx_read_phase_table_relation<FF>;
+using lookup_read_phase_length_relation = bb::avm2::lookup_tx_read_phase_length_relation<FF>;
+using lookup_read_public_call_request_relation = bb::avm2::lookup_tx_read_public_call_request_phase_relation<FF>;
+using lookup_jump_on_revert_relation = bb::avm2::lookup_tx_phase_jump_on_revert_relation<FF>;
+
+// using lookup_dispatch_exec_start_relation = bb::avm2::lookup_tx_dispatch_exec_start_relation<FF>;
+// using lookup_dispatch_exec_get_revert_relation = bb::avm2::lookup_tx_dispatch_exec_get_revert_relation<FF>;
+
+using lookup_read_tree_insert_value_relation = bb::avm2::lookup_tx_read_tree_insert_value_relation<FF>;
+using lookup_write_tree_insert_value_relation = bb::avm2::lookup_tx_write_tree_insert_value_relation<FF>;
+using lookup_read_l2_l1_msg_relation = bb::avm2::lookup_tx_read_l2_l1_msg_relation<FF>;
+using lookup_write_l2_l1_msg_relation = bb::avm2::lookup_tx_write_l2_l1_msg_relation<FF>;
+
+TEST(TxExecutionConstrainingTest, Basic)
+{
+    // clang-format off
+    TestTraceContainer trace({
+         {{ C::execution_sel, 1 }, { C::execution_pc, 0 }},
+         {{ C::execution_sel, 1 }, { C::execution_pc, 20 }, { C::execution_last, 1 }}
```

Comment:
probably you want to use tx and not execution here?

Reply:
ah good point - i actually thought this was the EmptyRow version, where the tx trace portion is empty. I'll change it to an actual EmptyRow test